src/Sylius/Bundle/CoreBundle/Tests/Form/Type/CatalogPromotion/CatalogPromotionActionTypeTest.php

```diff
@@ -0,0 +1,307 @@
+        $form->submit([
+            'type' => 'fixed_discount',
+            'configuration' => [
+                'WEB_US' => [
+                    'amount' => 'Not valid amount'
+                ]
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('fixed_discount', $catalogPromotionAction->getType());
+        $this->assertSame(['WEB_US' => ['amount' => null]], $catalogPromotionAction->getConfiguration());
+    }
+
+    /** @test */
+    public function it_updates_percentage_discount_with_not_valid_amount(): void
+    {
+        $percentageDiscount = $this->setupPercentageDiscount();
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $percentageDiscount);
+        $form->submit([
+            'type' => 'percentage_discount',
+            'configuration' => [
+                'amount' => 'Not valid amount'
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('percentage_discount', $catalogPromotionAction->getType());
+        $this->assertSame(['amount' => null], $catalogPromotionAction->getConfiguration());
+    }
+
+    protected function setUp(): void
+    {
+        $this->channelRepository = $this->prophesize(ChannelRepositoryInterface::class);
+
+        $currency = $this->prophesize(CurrencyInterface::class);
+        $currency->getCode()->willReturn('$');
+
+        $channel = $this->prophesize(ChannelInterface::class);
+        $channel->getCode()->willReturn('WEB_US');
+        $channel->getName()->willReturn('United States');
+        $channel->getBaseCurrency()->willReturn($currency->reveal());
+        $this->channel = $channel;
+
+        parent::setUp();
+    }
+
+    protected function getExtensions(): array
+    {
+        $catalogPromotionActionType = new CatalogPromotionActionType(
+            CatalogPromotionActionInterface::class,
+            ['sylius'],
+            [
+                'percentage_discount' => new PercentageDiscountActionConfigurationType(),
+                'fixed_discount' => new ChannelBasedFixedDiscountActionConfigurationType()
+            ]
+        );
+
+        $channelCollectionType = new ChannelCollectionType(
+            $this->channelRepository->reveal()
+        );
+
+        $validator = Validation::createValidatorBuilder()->getValidator();
+
+        return [
+            new PreloadedExtension([$catalogPromotionActionType, $channelCollectionType], []),
+            new ValidatorExtension($validator)
```

Comment:
Do we need this validator extension to be used here? I don't see any validation scenarios in this test. (of course, if you wish, we may add validation unit tests as well)
```suggestion
```